lib/rust/parser/engine/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+#![feature(test)]
+#![deny(unconditional_recursion)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+
+//! This library exposes the specialized version of enso lexer.
+//! Its sole purpose is to avoid getting lexer definition out of sync with its implementation
+//! (the generated engine), which requires the engine to live in separate crate.
+//!
+//! This separation enables a call to `lexer.specialize` (function that generates the lexer engine)
+//! inside `build.rs` during compilation. Its output is then stored in a new file `lexer-engine.rs`
+//! and exported by `lexer.rs`.
```

Comment:
```suggestion
//! This library exposes the specialized version of the enso lexer.
//!
//! Its sole purpose is to avoid the lexer definition getting out of sync with its implementation
//! (the generated engine), which requires the engine to live in a separate crate.
//!
//! This separation enables a call to `lexer.specialize` (the function that generates the lexer engine)
//! inside `build.rs` during compilation. Its output is then stored in a new file `lexer-engine.rs`
//! and exported by `lexer.rs`.
```

Please remember, doc comments should have a short, single-sentence header, then paragraphs explaining things. Paragraphs require a blank line as this is markdown.

The note about `lexer.specialize` should have a TODO comment noting that this should link once that function exists.